src/components/BadgeComponent.vue

```diff
@@ -0,0 +1,17 @@
+<template>
+  <span class="badge" :class="'badge-' + type">
+    <slot></slot>
+  </span>
+</template>
+
+<script lang="ts">
+import { Component, Vue } from "vue-property-decorator";
+
+@Component({
+  props: {
+    type: String,
```

Comment:
This is awesome, however to avoid clash of terms please replace type with variant